### PR TITLE
Add application architecture diagram

### LINKED ISSUE #1041: Add application architecture diagram to readme

### PR DESCRIPTION
**Pull request by dhmacs**
closes #1041 

qa_req 0

This PR updates the readme and adds an application architecture diagram

image

cc @ardelato

### CODE REVIEW

**Comment by djmetzle** [comment]
There is quite a bit incorrect about this chunk of the diagram, @dhmacs 
image

**Comment by djmetzle** [comment]
Or is that a speculative plan for improvements on the strapi deployment?

**Comment by dhmacs** [comment]
> There is quite a bit incorrect about this chunk of the diagram
> Or is that a speculative plan for improvements on the strapi deployment?

The purpose is to get a clear picture about how the production `react-commerce` works, and Strapi is part of the picture.
I don't have access to details around how Strapi is deployed, I've inferred this from a discussion with @danielbeardsley, but I've probably misunderstood parts of it.
Can you help me to get this right? What should I change about it?

@djmetzle

File: README.md [hunk 1]
```diff
@@ -46,49 +28,71 @@ Copy `.env.local.example` to `.env.local` and fill in the values.
...
+
+The `frontend` directory is structured as follows:
+
+-  `pages`: contains the Next.js routes (usually these just export code from a template)
+-  `models`: contains business logic (e.g. how to fetch product list from API, how to subscribe to newsletter, etc.)
+-  `templates`: you can think of these as view controllers. Templates are especially useful to handle multiple routes (e.g. product list template is used by several routes, like `/Parts`, `/Tools`, `/Parts/Mac` etc.).
+-  `layouts`: contains the layouts that each page can use. Right now we only have a default layout, but in the future we might have pages that will require to use a different layout component.
+-  `components`: contains the React view components
+-  `helpers`: contains reusable custom app-related helper functions
+-  `lib`: contains custom libraries that can stand on their own (e.g. `lib/algolia`). Think of these as packages that potentially could be used in other projects.
```
Comment (masonmcelvain) [inline README.md:85]:
```suggestion
-  `lib`: contains custom libraries that can stand on their own (e.g. `lib/strapi-sdk`). Think of these as packages that potentially could be used in other projects.
```

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 The diagram is awesome! Minor comment about a non-existent folder, but the meaning of the bullet is still conveyed so let's just merge as is.

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

**Comment by dhmacs** [comment]
deploy_block 🛑 on waiting for @djmetzle feedback. Also note to self: add Algolia

**Comment by danielbeardsley** [comment]
> There is quite a bit incorrect about this chunk of the diagram

This diagram aims to document the services that react-commerce talks to. There are many details left out that aren't that relevant here. I'll list a few of the things I can think of:

* "API"  and "Web" are not separate applications in the iFixit infra just different url routes inside the app
*  The `ALB` in front of strapi is also in front of our php app, though for strapi it is effectively a proxy cause there's only one strapi instance
* react-commerce requests to the php-app API actually go through cloudfront cause they happen over the public internet 

To me, these parts aren't critical to get exactly right cause the detail is fractal here and it'll never be a perfect representation. Are there other things you'd change @djmetzle?

**Comment by djmetzle** [comment]
That top right corner looks a lot more accurate now @dhmacs, good stuff!

**Comment by dhmacs** [comment]
un_deploy_block 🟢

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

**Pull request merged**